src/test/java/teammates/test/cases/ui/browsertests/InstructorFeedbackEditPageUiTest.java

```diff
@@ -284,6 +285,61 @@ private void testEditQuestionAction() throws Exception {
         feedbackEditPage.clickAndCancel(feedbackEditPage.getCancelQuestionLink(-1));
     }
     
+    private void testEditVisibilityAction() throws Exception {
+
+        ______TS("edit question 1 to Students-to-OwnTeammatesIncludingSelf");
+
+        feedbackEditPage.clickVisibilityOptionsForQuestion1();
+        feedbackEditPage.selectGiverTypeForQuestion1("Students in this course");        
+        feedbackEditPage.selectRecipientTypeForQuestion1("Giver's team members and Giver");
+        
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackQuestionEditToStudentsToOwnTeammatesIncludingSelf.html");
+
+        
+        ______TS("test visibility options of question 1");
+        feedbackEditPage.clickquestionSaveForQuestion1();
+        feedbackEditPage.clickVisibilityOptionsForQuestion1();
+        
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackQuestionVisibilityOptionsForOwnTeammatesIncludingSelf.html");
+        
+        
+        ______TS("test visibility preview of question 1");
+        feedbackEditPage.clickVisibilityPreviewForQuestion1();
+        WebElement visibilityMessage = browser.driver.findElement(By.id("visibilityMessage-1"));
+        feedbackEditPage.waitForElementVisibility(visibilityMessage);
+
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackQuestionVisibilityPreviewForOwnTeammatesIncludingSelf.html");
```

Comment:
Personally, I think it will be better if you can perform this check without resorting to full page html comparisons. It can be explicit here what you are testing for. Just check whether the rows in the table are displayed or not?